### PR TITLE
LOG-3432: Replace existing rules with rate function and add new alert for success log sync events

### PR DESCRIPTION
**Pull request by vparfonov**
### Description
This PR replaces the existing rules that divide the number of error events by the total number of events in order to fire an alert. Since the total number of events is much higher than the number of error events, this alert was never fired. Instead, it will use a `rate function` that will monitor the rate of error events over time and trigger the alert if the rate exceeds a certain threshold.

-  CollectorHighErrorRate
```
alert: CollectorHighErrorRate
    annotations:
      message: |-
        {{ $value }}% of records have resulted in an error by vector {{ $labels.instance }} component.
      summary: "Vector component errors are high"
    expr: |
         sum by(pod, instance)(rate(vector_component_errors_total[2m])) > 0.05
    for: 5m
    labels:
      severity: warning
      namespace: "openshift-logging"
```

- CollectorVeryHighErrorRate
```
alert: CollectorVeryHighErrorRate
    annotations:
      message: |-
        {{ $value }}% of records have resulted in an error by vector {{ $labels.instance }} component.
      summary: "Vector component errors are very high"
    expr: |
      sum by(pod, instance)(rate(vector_component_errors_total[2m])) > 0.2
    for: 5m
    labels:
      severity: critical
      namespace: "openshift-logging"
```

In addition to this change, I have also added a new alert that checks the number of success log sync events and fires an alert if it is zero. This will help to ensure that any issues with log synchronization.

- CollectorSyncFailed
```
alert: CollectorSyncFailed
    annotations:
      message: |-
        Vector collector {{ $labels.instance }} log syncing failed.
      summary: "Vector component errors"
    expr: |
         (sum by (pod, instance, component_name) (rate(vector_events_out_total{component_kind="sink"}[2m]))) == 0
    for: 5m
    labels:
      severity: critical
      namespace: "openshift-logging"
```

/cc 
/assign 

/cherry-pick 

### Links

- Depending on PR(s):
- Bugzilla:
- Github issue:
- JIRA:
- Enhancement proposal:

### CODE REVIEW

**Comment by jcantrill** [comment]
@vparfonov do we have any graphs which make this alert more obvious as to when it is triggered?

File: internal/metrics/alerts/vector.go [hunk 1]
```diff
@@ -21,27 +21,30 @@ const VectorPrometheusAlerts = `
         {{ $value }}% of records have resulted in an error by vector {{ $labels.instance }} component.
       summary: "Vector component errors are high"
     expr: |
-      100 * (
-          sum by(pod, instance)(rate(vector_component_errors_total[2m]))
-        /
-          sum by(pod, instance)(rate(vector_component_received_events_total[2m]))
-        ) > 10
-    for: 15m
+         sum by(pod, instance)(rate(vector_component_errors_total[2m])) > 0.05
```
[the remark this answers is not in the document]
Comment (vparfonov) [reply internal/metrics/alerts/vector.go:24]: I did it in empirical way, just trying several times and checking the value. I will add graph for this alert bellow

**Comment by vparfonov** [comment]
> @vparfonov do we have any graphs which make this alert more obvious as to when it is triggered?

alert01

at 17:38 all storage pods was shutdown

**Comment by jcantrill** [comment]
It's an improvement.  @alanconway do you think there is a way to document and quantify the threshold in a way we can expose in an API to allow tuning

**Comment by vparfonov** [comment]
@alanconway ping ^^^

**Review by simonpasquier — COMMENTED** [review]
2 general comments:
* you want to group by namespace too so the alerts always get a namespace label which is handy for users to understand which component is involved and where to route alerts
* I'd drop the instance label as the pod name is already there and probably more useful.

**Comment by alanconway** [comment]
As per slack conversation, no need for tuning, we should just verify the threshold is reasonable according to these pointers (it may already be):

> > [11:34](https://redhat-internal.slack.com/archives/C05E2657UAJ/p1687793653014219)
>  hey! in general OpenShift components don't provide ways to tune their alerting rules so you want to err on the safe side and make sure that you're not over-alerting. In 4.14, users will be able to "replace" their own alerts on top of the out-of-the-box alerts ("replace" = create new alert definitions and drop the oob alerts from being sent to Alertmanager).
> The main knobs (which can combine) to avoid alerting noise are:
> 
>     a generous threshold value
>     a large for value, typically warning alerts can wait for 15m before firing.
> 
> if you want to go fancy, you can implement SLO-type alerts ala Google SRE book but it's way more work.
> some general information is available there:
> https://github.com/openshift/enhancements/blob/master/enhancements/monitoring/alerting-consistency.md#documentation-required
>

**Review by simonpasquier — COMMENTED** [review]
> Since the total number of events is much higher than the number of error events, this alert was never fired. 

What about keeping (errors rate / total rate) and having a lower threshold value then?
I fear that the new approach has a similar issue when a low number of messages are  processed...
You can also have different rules to alert on the % of errors and on the absolute rate of errors.

**Comment by vparfonov** [comment]
> > Since the total number of events is much higher than the number of error events, this alert was never fired.
> 
> What about keeping (errors rate / total rate) and having a lower threshold value then? I fear that the new approach has a similar issue when a low number of messages are processed... You can also have different rules to alert on the % of errors and on the absolute rate of errors.

@simonpasquier Thank you for your review. Perhaps you are correct, although it is necessary for it to be **significantly** lower and have a shorter duration. I will try to experiment

**Pull request closed without merging**